client/components/sidebar.vue

```diff
@@ -101,16 +174,53 @@
        * the comments text.
       */
       showComments(comments) {
-        const currentComments = comments
-        .filter((c) => {
-          return !c.versionTo;
+        let currentComments = comments
+        .filter((comment) => {
+          return !comment.versionTo;
         });
 
         if (!currentComments.length) {
           this.documentComments = currentComments;
           return;
         }
 
+        const replies = currentComments.filter((comment) => {
+          return comment.replyTo !== null;
+        });
+
+        const groupedReplies = _.groupBy(replies, (reply) => {
+          return reply.highlightKey;
+        });
+
+        currentComments = currentComments.filter((comment) => {
+          return comment.replyTo === null;
+        }).map((comment) => {
+          let commentReplies = groupedReplies[comment.highlightKey];
```

Comment:
So groups (or threads) should be based on the first comment, not highlight key. This is why we have `replyTo`. In the future we might allow multiple comments on the same highlight. Then the first comment will not have a `replyTo` set, but others will have it set to that first comment's ID.